m2cgen/interpreters/c/interpreter.py

```diff
@@ -15,6 +15,7 @@ class CInterpreter(ToCodeInterpreter,
 
     supported_bin_vector_num_ops = {
         ast.BinNumOpType.MUL: "mul_vector_number",
+        ast.BinNumOpType.DIV: "div_vector_number",
```

Comment:
It seems to me that introducing new linear algebra function is not directly related to this PR but produces a huge diff which hurts reviewability a lot. Can you please move code changes connected to the new function into a separate PR?

Reply:
Although it's not too obvious, this change to ensemble model assembler is required in order for a new heuristics to perform at its best on models of this kind. This is why I changed the logic in the first place.
Unfortunately while iterating on this issue I did a terrible job at splitting logical changes into their corresponding commits. At this point coming up with a separate PR requires some time investment.
I 100% agree that I should've done a better job at splitting these 2, but it wasn't that obvious while I was working on the issue. Majority of the code changes is a copy-paste of `div` functions in tests. If it's not too much trouble for you to review, then I'm going to ask to keep this PR as is.

Reply:
Otherwise the entire heuristics is only focused in the `java/interpreter.py` module + `ast.py` with expressions count logic.

Reply:
Looking at LightGBM RF implementation I realized that I did a stupidest update ever :D I will revert div updates now, since we can achieve the same behavior with MUL. This is what happens when your focus is too narrow.

Reply:
Should be much better now.

Reply:
All I wanted to achieve was instead of doing this (a lot of expressions)
```
(1 / coef * a + 1 / coef * b + ... + 1 / coef * n)
```
to do this:
```
(a + b + ... + n) * (1 / coef)
```
But as a result of neglecting basic algebra and the distributive law in particular I ended up implementing this crap:
```
(a + b + ... + n) / coef
```

Reply:
> This is what happens when your focus is too narrow.

Ahaha, I know what you mean! 😃